### ISSUE TITLE
Problem Number 2006 MYSQL server has gone away when upload file with big

### ISSUE THREAD

**Issue by iFeras93** [avramovic/DMLite-ORM#8]
![whatsapp image 2016-11-07 at 3 58 16 pm](https://cloud.githubusercontent.com/assets/6824682/20078114/ee95e372-a547-11e6-8996-006c035d84d8.jpeg)

Problem Number 2006 MYSQL server has gone away when upload file with big size

the file is uploaded but the problem here data can't save in database.

plz, help me.

**Comment by avramovic** [comment]
Are you using the latest version? Because I can't see anything related to database on [line 448](https://github.com/avramovic/DMLite-ORM/blob/master/application/core/MY_Model.php#L448), so that line can't cause the MySQL error you are getting.

Can you show your code - how are you importing? How big is the file which is being imported? What is its format?

**Comment by iFeras93** [comment]
Sorry i write some code on MY_Model .. but this pic from the latest version

![screenshot 2016-11-08 01 59 49](https://cloud.githubusercontent.com/assets/6824682/20081263/12036adc-a557-11e6-9fe8-62168bdc8e6a.png)

My Code is Here : 
[code.txt](https://github.com/avramovic/DMLite-ORM/files/576970/code.txt)

the format of file is .ipa

**Comment by iFeras93** [comment]
When i upload file size <= 20M all things alright,
but file size > 20M show this problem

**Comment by avramovic** [comment]
I don't think this is directly related to DMLite ORM. Your MySQL connection times out probably because it's waiting too long for the file to be uploaded. Please take a look at this article: [MySQL error 2006: mysql server has gone away](http://stackoverflow.com/questions/7942154/mysql-error-2006-mysql-server-has-gone-away)

TL;DR: Check your MySQL `max_allowed_packet` and `wait_timeout` settings and try to increase them. I think while the script waits for the file to be uploaded, your MySQL connection times out. Thus, you should probably increase `wait_timeout` setting of your MySQL server.

**Comment by iFeras93** [comment]
Thank you  Mr. Avramovic 
The  problem is in mysql.connect_timeout and solve from server
